test/WebJobs.Script.Tests/HostFunctionMetadataProviderTests.cs

```diff
@@ -31,14 +33,24 @@ public HostFunctionMetadataProviderTests()
         [Fact]
         public void ReadFunctionMetadata_Succeeds()
         {
+            var testLoggerProvider = new TestLoggerProvider();
+            var loggerFactory = new LoggerFactory();
+            loggerFactory.AddProvider(testLoggerProvider);
+            var logger = loggerFactory.CreateLogger<HostFunctionMetadataProvider>();
             string functionsPath = Path.Combine(Environment.CurrentDirectory, @"..", "..", "..", "..", "..", "sample", "node");
             _scriptApplicationHostOptions.ScriptPath = functionsPath;
             var optionsMonitor = TestHelpers.CreateOptionsMonitor(_scriptApplicationHostOptions);
-            var metadataProvider = new HostFunctionMetadataProvider(optionsMonitor, NullLogger<HostFunctionMetadataProvider>.Instance, _testMetricsLogger, SystemEnvironment.Instance);
+            var metadataProvider = new HostFunctionMetadataProvider(optionsMonitor, logger, _testMetricsLogger, SystemEnvironment.Instance);
             var workerConfigs = TestHelpers.GetTestWorkerConfigs();
 
             Assert.Equal(18, metadataProvider.GetFunctionMetadataAsync(workerConfigs, SystemEnvironment.Instance, false).Result.Length);
+            var traces = testLoggerProvider.GetAllLogMessages();
             Assert.True(AreRequiredMetricsEmitted(_testMetricsLogger));
+
+            // Assert that the logs contain the expected messages
+            Assert.Equal(2, traces.Count);
+            Assert.Equal("Reading functions metadata (Host)", traces[0].FormattedMessage);
+            Assert.Equal("18 functions found (Host)", traces[1].FormattedMessage);
```

Comment:
Why don't we get the "x functions loaded" log here?

Reply:
The "x functions loaded" only happens in the `FunctionMetadataManager` [here](https://github.com/Azure/azure-functions-host/pull/9464/files#diff-08e726fbbb6cad2322ba3a945fc8002af0346b11566a5b94042e00f33ec9a592R91), so the host doesn't log it